File: training/php/src/Domain/Model/CreatureId.php
<?php

declare(strict_types=1);

namespace De\Sots\CellarsAndCentaurs\Domain\Model;

final class CreatureId
{
    public function __construct(public readonly string $id)
    {
    }

    public function idAsString(): string
    {
        return $this->id;
    }
}

File: training/ruby/lib/de/sots/cellars_and_centaurs/domain/model/creature_id.rb
# frozen_string_literal: true

require_relative '../../version'

module De::Sots::CellarsAndCentaurs::Domain::Model
  class CreatureId
    attr_reader :id

    def initialize(id)
      @id = id
    end

    def ==(other)
      other.is_a?(CreatureId) && other.id == id
    end
    alias eql? ==

    def hash
      id.hash
    end
  end
end
